ext-src/models/IqComponentModel.ts

```diff
@@ -50,62 +50,84 @@ export class IqComponentModel implements ComponentModel {
       return new Promise((c, e) => "my stubbed content entry");
     }
   
-    public async evaluateComponents() {
+    public evaluateComponents(): Promise<any> {
       console.debug("evaluateComponents");
-      await this.performIqScan();
+      return this.performIqScan();
     }
   
-    private async performIqScan() {
-      try {
-        let componentContainer = new ComponentContainer(this.requestService);
+    private async performIqScan(): Promise<any> {
+      return new Promise((resolve, reject) => {
+        try {
+          let componentContainer = new ComponentContainer(this.requestService);
 
-        let data: any;
+          window.withProgress(
```

Comment:
This was very fun to figure out, but after playing with it, I love it! What a weird experience! I had to use the .then() on this Thenable, because errors were getting swallowed by `withProgress)`. This actually makes the code a bit more readable than it was before (at least in my opinion)